propnet/core/models.py

```diff
@@ -194,13 +193,32 @@ def evaluate(self, symbol_quantity_dict, allow_failure=True):
         if not self.check_constraints({**symbol_value_dict, **out}):
             return {"successful": False,
                     "message": "Constraints not satisfied"}
+
         provenance = ProvenanceElement(
             model=self.name, inputs=list(symbol_quantity_dict.values()))
         out = self.map_symbols_to_properties(out)
         for symbol, value in out.items():
             try:
-                out[symbol] = Quantity(symbol, value, self.unit_map.get(symbol),
+                quantity = Quantity(symbol, value, self.unit_map.get(symbol),
                                        provenance=provenance)
+
+                if quantity.symbol.category in ('property', 'condition'):
```

Comment:
This is generally very solid; you might want to put these lines (i. e. 203-212 into a function or method (perhaps bound to Quantity) just to use less real estate in Model.evaluate.